test/framework/vmoperator/vmoperator.go

```diff
@@ -322,47 +294,49 @@ func ReconcileDependencies(ctx context.Context, c client.Client, config *Depende
 	}
 
 	// Create VirtualMachineClass in K8s and bind it to the user namespace
-	// TODO: figure out if to add more vm classes / if to make them configurable via config
-	vmClass := &vmoprv1.VirtualMachineClass{
-		ObjectMeta: metav1.ObjectMeta{
-			Name: config.UserNamespace.VirtualMachineClass,
-		},
-		Spec: vmoprv1.VirtualMachineClassSpec{
-			Hardware: vmoprv1.VirtualMachineClassHardware{
-				Cpus:   8,
-				Memory: resource.MustParse("64G"),
+	for _, vmc := range config.Spec.VirtualMachineClasses {
+		vmClass := &vmoprv1.VirtualMachineClass{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: vmc,
+			},
+			Spec: vmoprv1.VirtualMachineClassSpec{
+				// TODO: figure out if to make vm class configurable via API
+				Hardware: vmoprv1.VirtualMachineClassHardware{
+					Cpus:   8,
+					Memory: resource.MustParse("64G"),
```

Comment:
yeah, the value doesn't really matter for vcsim
BTW in case of vmc, I think we should have some pre-existing classes, and thus the value here becomes not relevant